cmd/apiProviderDownload_test.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProviderDownloadTestUtilsBundle struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func apiProviderDownloadMockUtilsBundle() *apiProviderDownloadTestUtilsBundle {
+	utilsBundle := apiProviderDownloadTestUtilsBundle{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return &utilsBundle
+}
+
+//Successful API Provider download cases
+func TestApiProviderDownloadSuccess(t *testing.T) {
+	t.Parallel()
+	t.Run("Successful Download of API Provider", func(t *testing.T) {
+		apiServiceKey := `{
+			"oauth": {
+				"url": "https://demo",
+				"clientid": "demouser",
+				"clientsecret": "******",
+				"tokenurl": "https://demo/oauth/token"
+			}
+		}`
+
+		config := apiProviderDownloadOptions{
+			APIServiceKey:   apiServiceKey,
+			APIProviderName: "provider1",
+		}
+		// test
+		httpClient := httpMockCpis{CPIFunction: "APIProviderDownload", ResponseBody: ``, TestType: "Positive"}
+		utilsMock := apiProviderDownloadMockUtilsBundle()
+		err := runApiProviderDownload(&config, nil, &httpClient, utilsMock)
+
+		if assert.NoError(t, err) {
+			t.Run("Assert file download", func(t *testing.T) {
+				assert.Equal(t, err, nil)
```

Comment:
This test is no different than the NoError check above. Please rather do a proper test of what was written to file, at the configured `DownloadPath`. Here would be an example of us checking a written file: https://github.com/SAP/jenkins-library/blob/master/cmd/mtaBuild_test.go#L117-L127

Reply:
Added both positive & Negative test for asserting file write. Thanks for your support.